scripts/version-linter.py

```diff
@@ -250,18 +269,18 @@ def assert_commit(commit, desc):
 
   subprocess.run(['git','fetch'],capture_output=False)
 
-  base_branch_commit=branch_commit(base_branch)
-  download_type_shapes('base',base_branch,base_branch_commit)
+  base_branch_commit = find_latest_type_shape_ref_on(base_branch,n=10)
```

Comment:
Why is this change introduced?

What if we pick the wrong commit, one that actually has a different type shape from the last branch commit?

Reply:
We are usually looking in mainline branches for reference so there is a silent agreement that if PR with changes in version linter was merged to compatible or develop it is a correct one.

Reply:
> PR with changes in version linter was merged to compatible or develop it is a correct one

Yes, but what if a PR changed type shape in the mainline branch for good, and we're reverting it here by accident?

It seems to me as if policy to only check against the last commit would be much clearer, inviting less unexpected outcomes so to say.